dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java

```diff
@@ -110,32 +78,53 @@
+                .build().requiredFields());
         interfaceTypes.put("FILEASSET", createInterfaceType(FILE_INTERFACE_NAME, fileAssetFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> pageAssetFields = new HashMap<>(contentFields);
+        addBaseTypeFields(pageAssetFields, ImmutablePageContentType.builder().name("dummy")
+                .build().requiredFields());
         interfaceTypes.put("HTMLPAGE", createInterfaceType(PAGE_INTERFACE_NAME, pageAssetFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> personaFields = new HashMap<>(contentFields);
+        addBaseTypeFields(personaFields, ImmutablePersonaContentType.builder().name("dummy")
+                .build().requiredFields());
         interfaceTypes.put("PERSONA", createInterfaceType(PERSONA_INTERFACE_NAME, personaFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> widgetFields = new HashMap<>(contentFields);
+        addBaseTypeFields(widgetFields, ImmutableWidgetContentType.builder().name("dummy")
+                .build().requiredFields());
         interfaceTypes.put("WIDGET", createInterfaceType(WIDGET_INTERFACE_NAME, widgetFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> vanityUrlFields = new HashMap<>(contentFields);
+        addBaseTypeFields(vanityUrlFields, ImmutableVanityUrlContentType.builder().name("dummy")
+                .build().requiredFields());
         interfaceTypes.put("VANITY_URL", createInterfaceType(VANITY_URL_INTERFACE_NAME, vanityUrlFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> keyValueFields = new HashMap<>(contentFields);
+        addBaseTypeFields(keyValueFields, ImmutableKeyValueContentType.builder().name("dummy")
+                .build().requiredFields());
         interfaceTypes.put("KEY_VALUE", createInterfaceType(KEY_VALUE_INTERFACE_NAME, keyValueFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> formFields = new HashMap<>(contentFields);
-        interfaceTypes.put("FORM", createInterfaceType(FORM_INTERFACE_NAME, formFields, new ContentResolver()));
+        addBaseTypeFields(formFields, ImmutableFormContentType.builder().name("dummy")
+                .build().requiredFields());
+        interfaceTypes.put("FORM", createInterfaceType(FORM_INTERFACE_NAME, formFields,
+                new ContentResolver()));
 
         final Map<String, TypeFetcher> dotAssetFields = new HashMap<>(contentFields);
         interfaceTypes.put("DOTASSET", createInterfaceType(DOTASSET_INTERFACE_NAME, dotAssetFields, new ContentResolver()));
     }
 
-    public static Map<String, TypeFetcher> getContentFields() {
-        return contentFields;
+    private static void addBaseTypeFields(Map<String, TypeFetcher> baseTypeFields,
+            List<Field> requiredFormFields) {
+        for (final Field formField : requiredFormFields) {
+            if(!formField.fixed()) continue;
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)